### PR TITLE
[#9915] Auto-sync API request type: Intent

### LINKED ISSUE #9915: Auto-sync API request type: Intent
Just realize we could do something further beyond #9196 

There is an `Intent` class in the backend (JAVA) and `intent.ts` interface and in the frontend (TypeScript). They have the same structure and we should make use of the `./gradlew generateTypes` to auto-sync them.

Note that `Intent.java` should be in `request` package.

### PR DESCRIPTION
**Pull request by savaliyagautam**
Fixes #9915

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -162,12 +162,22 @@ task generateApiRequestFormat(type: cz.habarta.typescript.generator.gradle.Gener
     outputFile = "src/web/types/api-request.ts"
 }
 
+task generateIntent(type: cz.habarta.typescript.generator.gradle.GenerateTask) {
```
Comment (xpdavid) [inline build.gradle]: You don't need to create a task. `generateApiRequestFormat` will help you. This indicates that you should move `Intent` to the correct folder.

File: build.gradle [hunk 2]
```diff
@@ -162,12 +162,22 @@ task generateApiRequestFormat(type: cz.habarta.typescript.generator.gradle.Gener
...
+    classes = [
+        "teammates.ui.webapi.action.Intent"
+    ]
+    mapEnum = "asEnum"
+    outputFile = "src/web/types/Intent.ts"
```
Comment (xpdavid) [inline build.gradle]: Instead, you should delete the `Intent.ts` and use `Intent` from `api-request.ts`

File: build.gradle (hunk 1 above)
Comment (savaliyagautam) [reply build.gradle]: @xpdavid Understood. Removed `generateIntent` task

File: build.gradle (hunk 2 above)
Comment (savaliyagautam) [reply build.gradle]: @xpdavid  Deleted `Intent.ts`

File: build.gradle (hunk 1 above)
Comment (savaliyagautam) [reply build.gradle]: @xpdavid Please review latest changes.

File: build.gradle [hunk 3]
```diff
@@ -157,7 +157,8 @@ task generateApiOutputFormat(type: cz.habarta.typescript.generator.gradle.Genera
 
 task generateApiRequestFormat(type: cz.habarta.typescript.generator.gradle.GenerateTask) {
     classPatterns = [
-        "teammates.ui.webapi.request.**"
+        "teammates.ui.webapi.request.**",
+        "teammates.ui.webapi.action.Intent"
```
Comment (xpdavid) [inline build.gradle]: Why not just move `Intent.java` into the request folder?

Comment (savaliyagautam) [reply build.gradle]: @xpdavid Moved `Intent.java` from action to request folder.

File: .gitignore [hunk 4]
```diff
@@ -35,6 +35,9 @@ test-output/*
 *.iml
 out/
 
+# VS code
+.vscode/settings.json
```
Comment (xpdavid) [inline .gitignore]: irrelevant changes?

Comment (savaliyagautam) [reply .gitignore]: Maybe. reverted changes for `gitignore`

**Pull request merged**